src/Profiles.ts

```diff
@@ -336,6 +344,32 @@ export class Profiles {
         return [updSession.ISession.user, updSession.ISession.password, updSession.ISession.base64EncodedAuth];
     }
 
+    public getAllTypes() {
+        return this.allTypes;
+    }
+
+    public async getNamesForType(type: string) {
+        const profileManager = await this.getCliProfileManager(type);
+        const profilesForType = (await profileManager.loadAll()).filter((profile) => {
+            return profile.type === type;
+        });
+        return profilesForType.map((profile)=> {
+            return profile.name;
+        });
+    }
+
+    public async directLoad(type: string, name: string): Promise<IProfileLoaded> {
+        let directProfile: IProfileLoaded;
+        const profileManager = await new CliProfileManager({
```

Comment:
Would it be possible to reuse `this.getCliProfileManager`? (Please forgive me if I'm missing something here)

Reply:
Good spot. Will update